src/java.base/share/man/java.md

```diff
@@ -1085,6 +1085,23 @@ These `java` options control the runtime behavior of the Java HotSpot VM.
 `-XX:CRaCRestoreFrom=`*directory*
 :   Restores from the specified checkpoint image.
 
+    When using this option, there is no requirement to specify a main class and its
+    arguments because the restored execution is based on the ones specified before
+    the checkpoint. However, it is possible to pass a new main class and new
+    arguments. After a successful restoration, the new main method will be invoked
+    with the specified arguments in the thread that started the checkpoint, on top
+    of its existing call stack.
+
+`-XX:+CRaCIgnoreRestoreIfUnavailable`
+:   If the checkpoint image specified for restoration is identified as being unusable,
+    continue with the normal startup instead of failing. Enabling this option does
+    not guarantee that the normal startup will follow any failed restoration attempt;
+    it only covers an early image/environment verification.
+
+    When restoring with this option, you should specify a main class and its
+    arguments as usual. If the restoration succeeds, they are ignored, but if it
+    fails, they are used for the normal startup process.
```

Comment:
Let's move the "may fail" note to the bottom. Suggested wording:

If the checkpoint image specified for restoration is identified as being unusable, continue with the normal startup instead of failing. When restoring with this option, you should specify command line arguments for a normal start-up process. If the restoration succeeds, arguments are ignored. Otherwise, they will be used for the JVM initialization. The JVM won't be attempt to start normally, if initial verification passes but restoration does not succeed.